sisl/viz/_plotables.py

```diff
@@ -1,264 +1,89 @@
 """
 This file provides tools to handle plotability of objects
 """
-
-from copy import copy
-from functools import wraps
+from sisl._dispatcher import ClassDispatcher, AbstractDispatch, ObjectDispatcher
 
 __all__ = ["register_plotable"]
 
 
-class PlotEngine:
-    """
-    Stores and takes care of calling all methods of a plotting engine.
-    """
-
-    def __init__(self, name, handler, default=None, methods={}):
-
-        self._raw_methods = {}
-        self._handler = handler
-
-        for name, method in methods.items():
-            self._add(name, method)
-
-        self._name = name
-        self._default = default
-
-    def copy(self, **kwargs):
-        return self.__class__(self._name, kwargs.get("handler", self._handler),
-            default=self._default, methods=self._raw_methods)
-
-    def __dir__(self):
-        return list(self._raw_methods)
-
-    def _add(self, name, function, default=False):
-        """
-        Makes a new function available to this plotting engine.
-
-        Parameters
-        -----------
-        name: str
-            the name of the plotting method. 
-
-            This will be the name of the attribute where the function is stored.
-        function: function
-            the plotting method.
-        default: bool, optional
-            whether this method is the default for the engine.
-
-            The default method can be called by simply calling the engine.        
-        """
-        if default:
-            self._default = name
-
-        self._raw_methods[name] = function
-
-    def _get_wrapped_function(self, name):
-        function = self._raw_methods[name]
-
-        # This is the function that really does the plotting and goes
-        # into the engine
-        @wraps(function)
-        def real_function(*args, **kwargs):
-            return function(self._handler._obj, *args, **kwargs)
+class ClassPlotHandler(ClassDispatcher):
+    """Handles all plotting possibilities for a class"""
 
-        return real_function
+    def __init__(self, *args, **kwargs):
+        kwargs["instance_dispatcher"] = ObjectPlotHandler
```

Comment:
Better to only add this in case user put it in.

Something like:
```python
if not "instance_dispatcher" in kwargs:
    kwargs["instance_dispatcher"] = ObjectPlotHandler
```
ensures users can do what-ever.

Reply:
Done :+1: